framework/console/controllers/MigrateController.php

```diff
@@ -319,6 +326,15 @@ protected function removeMigrationHistory($version)
 
     /**
      * @inheritdoc
+     * @since 2.0.13
+     */
+    protected function getMigrationNameLimit()
+    {
+        return static::MAX_NAME_LENGTH;
```

Comment:
What about detecting real limit from DB schema? People may want to use longer names and change length of this column in DB.

Reply:
Migrations are meant to be applied to DB connection defined so it's a bad idea. We don't know the environment beforehand.

Reply:
I'm generally okay with detecting it, but in `actionCreate` the migration table may not yet exist.

Reply:
@samdark Sorry don't get it. There is already check whether migration table exist:
https://github.com/cebe/yii2/blob/37204ebcb49f5ba96bf671687da50a5e27a7213b/framework/console/controllers/MigrateController.php#L208-L210

What is the problem in checking column length in the same way?

Anyway, this is BC break without it - if I changed column limit to 255 and have have migration with 230 chars, this change will break my environment. And it can't be easily fixed by configuration.

Reply:
good point.

Reply:
> I'm generally okay with detecting it, but in actionCreate the migration table may not yet exist.

Then we can safely use default limit.

Reply:
fixed. however create action needs database access now, not sure how much this change could break anything.